### ISSUE TITLE
iOS: Carry contextual information for dates, times, and numbers

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30890]
## Requires API Proposal
This issue may require a new API, or a change to an existing API. An API proposal should be added and discussed before proceeding with implementation. The API proposal can be added in the comments of this issue or linked as a separate issue.

## Description
Dates and Numbers can be very difficult to correctly contextualize when screen readers try to vocalize the text. It would be preferable if logic for dates and times were encapsulated so that custom logic for accessibility labels would not have to be created every time.

## React Native version:
v0.63

## Expected Behavior
One idea would be pseudo code like this:
```
<Text type="date" style={...}>
 01/12/1990
</Text>
```
VoiceOver would read (for a US locale): "January twelfth, nineteen ninety."

## iOS Details
There are particular classes on native iOS that support this kind of encapsulation and context automatically like `NSDateFormatter`, `NSDateComponents`, and `NSNumberFormatter`.
